sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -150,6 +151,30 @@ abstract class JdbcDialect extends Serializable {
   def beforeFetch(connection: Connection, properties: Map[String, String]): Unit = {
   }
 
+  /**
+   * Do some extra properties validation work in addition to the validation in [[JDBCOptions]],
+   * eg. [[validateFetchSize()]].
+   * @param properties The connection properties.  This is passed through from the relation.
+   */
+  def validateProperties(properties: Map[String, String]): Unit = {
+    val fetchSize = properties.getOrElse(JDBCOptions.JDBC_BATCH_FETCH_SIZE, "0").toInt
+    validateFetchSize(fetchSize)
+  }
+
+  /**
+   * This is to validate the fetch size specified via [[JDBCOptions.JDBC_BATCH_FETCH_SIZE]].
+   * The implementation here requires the fetch size >= 0, sub classes may override it to meet
+   * requirement of different dialects.
+   * @param size the fetch size to validate.
+   */
+  protected def validateFetchSize(size: Int): Unit = {
```

Comment:
How about just making this IF public in JdbcDialect? Then, dropping `validateProperties`.

Reply:
I'd like to keep validateProperties here to allow more possible property validation added in it. 
By doing so, we expose only one public method (validateProperties) instead of one for each property validation. 
It's my thought, What do you think?

Reply:
Or, just overridding `validateProperties` in `MySQLDialect`?

Reply:
make sense, thanks.